src/precice/impl/DataContext.hpp

```diff
@@ -2,27 +2,82 @@
 
 #include <string>
 #include "MappingContext.hpp"
+#include "MeshContext.hpp"
 #include "mesh/SharedPointer.hpp"
 
 namespace precice {
 namespace impl {
 
 /**
- * @brief Stores one Data object with related context. If this dataContext is not associated with a mapping,
- * fromData and toData refer to the same data object.
+ * @brief Stores one Data object with related mesh.
+ *
+ * - If this dataContext is associated with a mapping, fromData and toData will be set correspondingly.
+ *   One of the two must be equal to providedData. fromData and toData must be different.
+ * - If this dataContext is not associated with a mapping, fromData and toData will be unset.
  */
-struct DataContext {
-  bool used = false;
+class DataContext {
 
-  std::string getName() const;
+public:
+  DataContext(mesh::PtrData data, mesh::PtrMesh mesh); // @ todo use mesh context for constructor? Would be more consistent with configureForReadMapping and configureForWriteMapping
```

Comment:
I will check whether this is easily doable or otherwise remove the todo.

Reply:
I tried to use meshContext, but it is not easy to do, because the meshContext is not available in the participant configuration. Calling `Participant::meshContext(meshID)` gave me a failing assertion in some tests. Therefore, I will remove this todo and leave the rest as it is.